test/e2e/certificates_test.go

```diff
@@ -171,10 +185,109 @@ var _ = Describe("ACME Certificate", Ordered, func() {
+
+			clusterIssuer := &certmanagerv1.ClusterIssuer{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      clusterIssuerName,
+					Namespace: ns.Name,
+				},
+				Spec: certmanagerv1.IssuerSpec{
+					IssuerConfig: certmanagerv1.IssuerConfig{
+						ACME: &v1.ACMEIssuer{
+							Server: "https://acme-staging-v02.api.letsencrypt.org/directory",
+							PrivateKey: certmanagermetav1.SecretKeySelector{
+								LocalObjectReference: certmanagermetav1.LocalObjectReference{
+									Name: "letsencrypt-dns01-issuer",
+								},
+							},
+							Solvers: []v1.ACMEChallengeSolver{
+								{
+									DNS01: &v1.ACMEChallengeSolverDNS01{
+										Webhook: &v1.ACMEIssuerDNS01ProviderWebhook{
+											GroupName:  "acme.borup.work",
+											SolverName: "ibmcis",
+											Config: &apiextensionsv1.JSON{
+												Raw: []byte(fmt.Sprintf(`{
+													"api-key-secret-ref": {
+														"local-object-reference": {
+															"name": "ibmcis-credentials"
+														},
+														"key": "api-token"
+													},
+													"cisCRN": ["%s"]
+												}`, cisCRN)),
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			}
+			_, err = certmanagerClient.CertmanagerV1().ClusterIssuers().Create(ctx, clusterIssuer, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			defer certmanagerClient.CertmanagerV1().ClusterIssuers().Delete(ctx, clusterIssuerName, metav1.DeleteOptions{})
+
+			By("creating new certificate")
+			certDomain := "adwa." + appsDomain // acronym for "ACME dns-01 Webhook Ambient", short naming to pass dns name validation
+			certName := "letsencrypt-cert"
+			cert := &certmanagerv1.Certificate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      certName,
+					Namespace: ns.Name,
+				},
+				Spec: certmanagerv1.CertificateSpec{
+					IsCA:       false,
+					CommonName: certDomain,
+					SecretName: certName,
+					DNSNames:   []string{certDomain},
+					IssuerRef: certmanagermetav1.ObjectReference{
+						Name: clusterIssuerName,
+						Kind: "ClusterIssuer",
+					},
+				},
+			}
+
+			_, err = certmanagerClient.CertmanagerV1().Certificates(ns.Name).Create(ctx, cert, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			defer certmanagerClient.CertmanagerV1().Certificates(ns.Name).Delete(ctx, certName, metav1.DeleteOptions{})
+
+			By("waiting for certificate to get ready")
+			err = waitForCertificateReadiness(ctx, certName, ns.Name)
+			Expect(err).NotTo(HaveOccurred())
+
+			By("checking for certificate validity from secret contents")
+			err = verifyCertificate(ctx, certName, ns.Name, certDomain)
+			Expect(err).NotTo(HaveOccurred())
+		})
 	})
 
 	Context("dns-01 challenge using ambient credentials", func() {
 		It("should obtain a valid LetsEncrypt certificate in AWS mint/passthrough cluster", func() {
+			if _, ok := os.LookupEnv(targetPlatformEnvironmentVar); ok {
```

Comment:
Ditto, my other above comment. For now, let's skip if Infrastructure's platform status isn't IBM cloud. In the future, we would want to modularize these test cases so we have a specific suite for IBM Cloud, AWS, GCP, etc. etc. (today we don't have a good way of differentiation, so stuck with `Skip()`).

Reply:
Right. Today, we use a UPI install on IBM Cloud to use with ppc64le machines.

Reply:
Last question, can someone run the same test if they're on a non-ppc64le IBM cloud OpenShift cluster. Probably, not right? Because PowerVS is generally handled differently from what's in s390x, amd64 IBM Cloud OpenShift clusters., ref: https://docs.openshift.com/container-platform/4.15/installing/installing_ibm_cloud_public/installing-ibm-cloud-customizations.html#installation-ibm-cloud-config-yaml_installing-ibm-cloud-customizations vs https://docs.openshift.com/container-platform/4.15/installing/installing_ibm_powervs/installing-ibm-power-vs-customizations.html#installation-ibm-power-vs-config-yaml_installing-ibm-power-vs-customizations. 

Essentially, I'm highlighting between `platform: ibmcloud` vs `platform: powervs` vs `platform: none`. Hope am not deflecting too much from the original context but my only intent here is to understand what the "Platform" view looks like from within the OpenShift cluster where you intend to run these test cases.

Reply:
Yes, we have one team running the tests with s390x on premises. The only part that doesn't work for them is the http01.